server/commands.go

```diff
@@ -0,0 +1,472 @@
+	}
+}
+
+func (p *Plugin) executeCommandIcebreakerShowProposals(args *model.CommandArgs) *model.CommandResponse {
+	sourceUser, _ := p.API.GetUser(args.UserId)
+	if !sourceUser.IsSystemAdmin() { //TODO: Check for Channel owner instead of System Admin
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         "You need to be admin in order to show proposed questions",
+		}
+	}
+
+	data := p.ReadFromStorage(args.TeamId, args.ChannelId)
+
+	if len(data.ProposedQuestions[args.TeamId][args.ChannelId]) == 0 {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         "There are no proposed questions for this channel...",
+		}
+	}
+
+	message := "Proposed questions:\n"
+	for index, question := range data.ProposedQuestions[args.TeamId][args.ChannelId] {
+		message = message + fmt.Sprintf("%d.\t%s:\t%s\n", index, question.Creator, question.Question)
+	}
+
+	return &model.CommandResponse{
+		ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		Text:         message,
+	}
+}
+
+func (p *Plugin) executeCommandIcebreakerShowApproved(args *model.CommandArgs) *model.CommandResponse {
+	data := p.ReadFromStorage(args.TeamId, args.ChannelId)
+
+	if len(data.ApprovedQuestions[args.TeamId][args.ChannelId]) == 0 {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         "There are no questions for this channel...",
+		}
+	}
+
+	message := "Questions:\n"
+	for index, question := range data.ApprovedQuestions[args.TeamId][args.ChannelId] {
+		message = message + fmt.Sprintf("%d.\t%s:\t%s\n", index, question.Creator, question.Question)
+	}
+
+	return &model.CommandResponse{
+		ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		Text:         message,
+	}
+}
+
+func (p *Plugin) executeCommandIcebreaker(args *model.CommandArgs) *model.CommandResponse {
+	data := p.ReadFromStorage(args.TeamId, args.ChannelId)
+
+	//check if there are any approved questions yet
+	if len(data.ApprovedQuestions[args.TeamId][args.ChannelId]) == 0 {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         "There are no approved questions that I can ask...",
+		}
+	}
+
+	//get a random user that is not a bot
+	user, err := p.GetRandomUser(args.ChannelId)
+	if err != nil {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         "There is no user I can ask a question for...",
+		}
+	}
+
+	//build the question and ask it
+	question := data.ApprovedQuestions[args.TeamId][args.ChannelId][rand.Intn(len(data.ApprovedQuestions[args.TeamId][args.ChannelId]))]
+	message := fmt.Sprintf("Hey @%s! %s", user.GetDisplayName(""), question.Question)
+	post := &model.Post{
+		ChannelId: args.ChannelId,
+		RootId:    args.RootId,
+		UserId:    p.botID,
+		Message:   message,
+	}
+	_, err = p.API.CreatePost(post)
+	if err != nil {
```

Comment:
It's generally preferred to handle error inline when possible. Here it can be handled inline.